wit/couchbase/subdocument.wit

```diff
@@ -0,0 +1,253 @@
+package wasmcloud:couchbase@0.1.0-draft;
+
+/// Perform sub-document lookup operations on Documents stored in a Couchbase cluster
+///
+/// Reference: https://docs.couchbase.com/go-sdk/current/howtos/subdocument-operations.html#retrieving
+interface subdocument-lookup {
+  use types.{
+    document-id, request-span, document, subdocument-path, document-error, retry-strategy
+  };
+
+  /// Options for performing a lookup
+  record lookup-get-options {
+    /// Whether the lookup refers to an extended attribute, rather than the document itself
+    xattr: option<bool>,
+  }
+
+  type lookup-exists-options = lookup-get-options;
+  type lookup-count-options = lookup-get-options;
+
+  /// Types of different lookup operations that can be specified
+  variant lookup-operation {
+    /// Whether the particular path exists
+    exists(tuple<subdocument-path, lookup-exists-options>),
+    /// Whether the particular path exists
+    get(tuple<subdocument-path, lookup-get-options>),
+    /// Whether the particular path exists
+    count(tuple<subdocument-path, lookup-count-options>),
+  }
+
+  /// Errors that occur during lookup operation
+  variant lookup-error {
+    /// A completely unexpected error
+    unexpected(string),
+  }
+
+  /// Options that control the lookup operation
+  record lookup-options {
+    /// Timeout that should be used, in milliseconds
+    /// If not specified, the implementer *may* provide a default timeout.
+    timeout-ms: option<u64>,
+
+    /// How and whether to retry the operation
+    retry-strategy: option<retry-strategy>,
+
+    /// A known span to associate this lookup with
+    parent-span: option<request-span>,
+  }
+
+  /// Errors that occur during lookup for an individual subdocument path
+  variant subdocument-lookup-error {
+    /// The provided path does not exist in the document
+    path-does-not-exist(string),
+    /// The path already exists (ex. during an insert)
+    path-already-exist(string),
```

Comment:
Does that make sense in the context of a subdoc lookup ? It's a read operation.

Reply:
So this actually is modeled after the [Subdocument query documentation](https://docs.couchbase.com/go-sdk/current/howtos/subdocument-operations.html#error-handling) -- it's specifically in the case of an insert that this is possible, I think! 

If I misread that section totally happy to take that out

Reply:
This definitely belongs in the `mutate-error` variant -- will move!

- [ ] move `path-already-exists` variant value to `subdocument-mutate-error`